opte/src/engine/port.rs

```diff
@@ -389,7 +386,7 @@ pub struct Port {
 macro_rules! check_state {
     ( $sg:expr, [ $( $state:expr ),* ] ) => {
         if $( *$sg != $state )&&* {
-            Err(OpteError::BadState)
+            Err(OpteError::BadState(format!("{}", *$sg)))
```

Comment:
Might not be a big deal, but I think we could leave the `OpteError::BadState` variant as carrying the actual state, rather than the stringification. That would make it easier to operate on the failure programmatically, though I don't know that's important yet.

Reply:
Yea, I thought about that. The problem is it then requires `PortState` to be part of the API. Which, maybe it should be, but I didn't want to go down that refactoring path with this already large PR.